lib/Network/OAuth2/Server/API.hs

```diff
@@ -169,42 +170,29 @@ processTokenRequest _ _ Nothing _ = do
     errorLog "processTokenRequest" "Checking credentials but none provided."
     invalidRequest "No credentials provided"
 processTokenRequest ref t (Just client_auth) req = do
-    (client_id, modified_scope) <- checkCredentials ref client_auth req
-    user <- case req of
-        RequestAuthorizationCode{..} -> do
-            -- Load the authorization code.
-            code' <- liftIO $ storeReadCode ref requestCode
-            case code' of
-                Nothing -> throwError $
-                    OAuth2Error InvalidGrant (preview errorDescription "This code is not valid.") Nothing
-                Just c  -> do
-                    -- TODO(thsutton) Revoke tokens if this code has already
-                    -- been used as described in:
-                    -- http://tools.ietf.org/html/rfc6749#section-4.1.2
-                    res <- liftIO $ storeDeleteCode ref requestCode
-                    case res of
-                        True  -> return $ Just (requestCodeUserID c)
-                        False -> throwError $ OAuth2Error InvalidRequest
-                                                          (preview errorDescription "This code is not valid.")
-                                                          Nothing
-        RequestClientCredentials{} -> return Nothing
-        RequestRefreshToken{..} -> do
-                -- Decode previous token so we can copy details across.
-                previous <- liftIO $ storeReadToken ref (Left requestRefreshToken)
-                case previous of
-                    Nothing -> return Nothing
-                    Just (tid, details) -> do
-                        let uid = tokenDetailsUserID details
-                        -- TODO(thsutton) Check the result here; better yet,
-                        -- refactor so we revoke *after* creating the new tokens.
-                        liftIO $ storeRevokeToken ref tid
-                        return uid
+    debugLog "checkCredentials" "Checking some credentials"
+    c_id <- checkClientAuth ref client_auth
+    client_id <- case c_id of
```

Comment:
Someone tell me a better way to write this. `maybe (error "Computer says no") (return)` is OK ish?